File: server/src/main/java/com/ringout/api/auth/dto/LoginResponse.java
package com.ringout.api.auth.dto;

public record LoginResponse(
        String accessToken,
        String refreshToken,
        String signupToken,
        boolean isNewUser
) {

    public static LoginResponse loggedIn(String accessToken, String refreshToken) {
        return new LoginResponse(accessToken, refreshToken, null, false);
    }

    public static LoginResponse signupRequired(String signupToken) {
        return new LoginResponse(null, null, signupToken, true);
    }
}

File: client/shared/src/commonMain/kotlin/com/joon/ringout/data/auth/remote/model/LoginResponse.kt
package com.joon.ringout.data.auth.remote.model

import kotlinx.serialization.Serializable

@Serializable
data class LoginResponse(
    val accessToken: String? = null,
    val refreshToken: String? = null,
    val signupToken: String? = null,
    val isNewUser: Boolean,
)
